File: flutter/lib/models/schedule_exception.dart
class ScheduleException {
  final int id;
  final int userId;
  final DateTime date;
  final bool isClosed;
  final String? openingTime;
  final String? closingTime;
  final String notes;
  final DateTime createdAt;
  final DateTime updatedAt;

  ScheduleException({
    required this.id,
    required this.userId,
    required this.date,
    required this.isClosed,
    this.openingTime,
    this.closingTime,
    required this.notes,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ScheduleException.fromJson(Map<String, dynamic> json) {
    return ScheduleException(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      date: DateTime.parse(json['date'] as String),
      isClosed: json['is_closed'] as bool,
      openingTime: json['opening_time'] as String?,
      closingTime: json['closing_time'] as String?,
      notes: json['notes'] as String? ?? '',
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'date': date.toIso8601String().split('T')[0],
      'is_closed': isClosed,
      'opening_time': openingTime,
      'closing_time': closingTime,
      'notes': notes,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}

File: golang/models/schedule_exception.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type ScheduleException struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	UserID      uint           `gorm:"not null;index" json:"user_id"`
	User        User           `gorm:"foreignKey:UserID" json:"-"`
	Date        time.Time      `gorm:"type:date;not null;index" json:"date"`
	IsClosed    bool           `gorm:"default:false" json:"is_closed"`
	OpeningTime *string        `json:"opening_time,omitempty"`
	ClosingTime *string        `json:"closing_time,omitempty"`
	Notes       string         `json:"notes"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

func (ScheduleException) TableName() string {
	return "schedule_exceptions"
}
